addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_action.xml

```diff
@@ -13,6 +13,9 @@
                     t-key="filter.id"
                 />
             </t>
+            <t t-set-slot="control-panel-navigation-additional">
+                <SpreadsheetShareButton t-if="status === Status.Loaded" t-key="activeDashboardId" model="state.activeDashboard.model" onSpreadsheetShared.bind="shareSpreadsheet"/>
```

Comment:
There's a functional issue. If you click on the share button, then update some filters, then click the share button again, the new filters values are not taken into account. (that being said, it's the same for regular spreadsheets, if you share, then edit, then share again: the first share is kept).

Maybe the `SpreadsheetShareButton` could check if the freezed data is different (or not) and create a new share (or not).